src/libprojectM/Renderer/Waveform.cpp

```diff
@@ -43,37 +42,38 @@ void Waveform::InitVertexAttrib() {
 }
 
 void Waveform::Draw(RenderContext &context)
- {
+{
     // scale PCM data based on vol_history to make it more or less independent of the application output volume
-    const float  vol_scale = context.beatDetect->getPCMScale();
-
-		// Make sure samples<=points.size().  We could reallocate points, but 512 is probably big enough.
-		size_t samples_count = this->samples;
-		if (samples_count > this->points.size())
-				samples_count = this->points.size();
+    const float vol_scale = context.beatDetect->getPCMScale();
 
-		float *value1 = new float[samples_count];
-		float *value2 = new float[samples_count];
-		context.beatDetect->pcm->getPCM( value1, samples_count, 0, spectrum, smoothing, 0);
-		context.beatDetect->pcm->getPCM( value2, samples_count, 1, spectrum, smoothing, 0);
+	// Make sure samples<=points.size().  We could reallocate points, but 512 is probably big enough.
+    size_t samples_count = this->samples;
+    if (samples_count > this->points.size())
+        samples_count = this->points.size();
 
-		float mult = scaling*( spectrum ? 0.015f :1.0f);
-#ifdef WIN32
-	std::transform(&value1[0], &value1[samples_count], &value1[0], bind2nd(std::multiplies<float>(), mult));
-	std::transform(&value2[0], &value2[samples_count], &value2[0], bind2nd(std::multiplies<float>(), mult));
-#else
-	std::transform(&value1[0], &value1[samples_count], &value1[0], std::bind(std::multiplies<float>(), std::placeholders::_1, mult));
-	std::transform(&value2[0], &value2[samples_count], &value2[0], std::bind(std::multiplies<float>(), std::placeholders::_1, mult));
-#endif /** WIN32 */
+    float *value1 = new float[samples_count];
+    float *value2 = new float[samples_count];
+    if (spectrum)
+    {
+        // TODO support smoothing parameter for getSpectrum()
+        context.beatDetect->pcm->getSpectrum( value1, 0, samples_count, 1.0 );
```

Comment:
Doing something like
```suggestion
        context.beatDetect->pcm->getSpectrum( value1, CHANNEL_L, samples_count, 1.0 );
```
Would be more readable IMO

Reply:
Yeah, I thought about changing that to an enum.